#!/usr/bin/env python
PACKAGE = "vineyard_midrow_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("line_detection_distance_threshold",    double_t, 0, "Line Detection Distance Threshold",           0.4,       0.05,       2)
gen.add("line_detection_max_angle",             double_t, 0, "Line Detection Distance Threshold",           10.0,       0.0,       90.0)
gen.add("line_detection_min_n_points",          int_t, 0, "Line Detection Minimum Number of Points",       100,       1,       1000)


gen.add("keep_box_lower_bound_x",             double_t, 0, "Keep Box Lower Bound X",                    -4.0,      -5,         5)
gen.add("keep_box_upper_bound_x",             double_t, 0, "Keep Box Upper Bound X",                    4.0,     -5,         5)

gen.add("keep_box_lower_bound_y",             double_t, 0, "Keep Box Lower Bound Y",                    -2.0,      -5,         0)
gen.add("keep_box_upper_bound_y",             double_t, 0, "Keep Box Upper Bound Y",                    2.0,     0,         5)

gen.add("keep_box_lower_bound_z",             double_t, 0, "Keep Box Lower Bound Z",                    0.0,      -5,         5)
gen.add("keep_box_upper_bound_z",             double_t, 0, "Keep Box Upper Bound Z",                    5.0,     0,         10)


gen.add("remove_box_lower_bound_x",             double_t, 0, "Remove Box Lower Bound X",                    -0.7,      -5,         5)
gen.add("remove_box_upper_bound_x",             double_t, 0, "Remove Box Upper Bound X",                    -0.25,     -5,         5)

gen.add("remove_box_lower_bound_y",             double_t, 0, "Remove Box Lower Bound Y",                    -0.8,      -5,         0)
gen.add("remove_box_upper_bound_y",             double_t, 0, "Remove Box Upper Bound Y",                    0.5,     0,         5)

gen.add("remove_box_lower_bound_z",             double_t, 0, "Remove Box Lower Bound Z",                    -0.3,      -5,         5)
gen.add("remove_box_upper_bound_z",             double_t, 0, "Remove Box Upper Bound Z",                    0.3,     -5,         5)


exit(gen.generate(PACKAGE, "vineyard_midrow_detection", "MidrowDetection"))